### PR TITLE
Added ToyoshimaHidenori's keymap for Ergodash mini

### PR DESCRIPTION
**Pull request by ToyoshimaHidenori**
## Description
Added  ToyoshimaHidenori's keymap for ergodash/mini

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>
+*/
+
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c:19]: This isn't actually needed.

```suggestion
```

File: keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _ADJUST 16
```
Comment (drashna) [inline keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c:24]:
```suggestion
enum layers {
    _QWERTY,
    _LOWER,
    _RAISE,
    _ADJUST,
};
```

File: keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  LOWER,
+  RAISE,
+  ADJUST,
```
Comment (drashna) [inline keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c:30]:
```suggestion
    QWERTY = SAFE_RANGE,
    LOWER,
    RAISE,
    ADJUST,
```

File: keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+};
+
+#define EISU LALT(KC_GRV)xx
```
Comment (drashna) [inline keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c:33]: I suspect that you didn't mean to have the `xx` here.

```suggestion
#define EISU LALT(KC_GRV)


```

File: keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+   * | Shift|   Z  |   X  |   C  |   V  |   B  |  GUI |                    |  GUI |   N  |   M  |   ,  |   .  |   /  | Shift|
+   * |-------------+------+------+------+------+------+------+------+------+------+------+------+------+------+-------------|
+   * | Ctrl |  ESC |  ALt | ESC  |||||||| Shift| Raise| Space|||||||| enter| Lower| Bcspc|||||||| Left | Down |  Up  | Right|
+   * ,----------------------------------------------------------------------------------------------------------------------.
+   */
+    [_QWERTY] = LAYOUT( \
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_MINS,                        KC_EQL , KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSLS, \
+    KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_LBRC,                        KC_RBRC, KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_LGUI,                        KC_RGUI, KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, \
+    KC_LCTL, KC_ESC,  KC_LALT, KC_ESC,           KC_LSFT, RAISE,   KC_SPC,        KC_ENT, LOWER,   KC_BSPC,          KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
Comment (drashna) [inline keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c:51]:
```suggestion
    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_MINS,                        KC_EQL , KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSLS,
    KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_LBRC,                        KC_RBRC, KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_LGUI,                        KC_RGUI, KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,
    KC_LCTL, KC_ESC,  KC_LALT, KC_ESC,           KC_LSFT, RAISE,   KC_SPC,        KC_ENT, LOWER,   KC_BSPC,          KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT 
```

File: keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+   * |-------------+------+------+------+------+------+------+------+------+------+------+------+------+------+-------------|
+   * | Ctrl |  ESC |  ALt | EISU |||||||| kana | Raise| Space|||||||| Enter| Lower|Delete|||||||| Home |PageDn|PageUp|  End |
+   * ,----------------------------------------------------------------------------------------------------------------------.
+   */
+  [_LOWER] = LAYOUT(
+    KC_TILD, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_UNDS,                       KC_PLUS, KC_6,    KC_7,    KC_8,   KC_9,    KC_0,    KC_PIPE, \
+    KC_LCTL, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_LCBR,                       KC_RCBR, KC_LEFT, KC_DOWN, KC_UP,  KC_RGHT, KC_COLN, KC_DQT , \
+    KC_LSFT, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_LGUI,                       KC_RGUI, KC_MINS, KC_EQL,  KC_LT,  KC_GT,   KC_QUES, KC_RSFT, \
+    KC_LCTL, KC_ESC, KC_LALT,  KC_LANG2,         KC_LANG1,RAISE,   KC_SPC,       KC_ENT, LOWER,   KC_DEL,           KC_HOME,KC_PGDN, KC_PGUP, KC_END   \
+  ),
```
Comment (drashna) [inline keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c:70]:
```suggestion
    KC_TILD, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_UNDS,                       KC_PLUS, KC_6,    KC_7,    KC_8,   KC_9,    KC_0,    KC_PIPE,
    KC_LCTL, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_LCBR,                       KC_RCBR, KC_LEFT, KC_DOWN, KC_UP,  KC_RGHT, KC_COLN, KC_DQT ,
    KC_LSFT, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_LGUI,                       KC_RGUI, KC_MINS, KC_EQL,  KC_LT,  KC_GT,   KC_QUES, KC_RSFT,
    KC_LCTL, KC_ESC, KC_LALT,  KC_LANG2,         KC_LANG1,RAISE,   KC_SPC,       KC_ENT, LOWER,   KC_DEL,           KC_HOME,KC_PGDN, KC_PGUP, KC_END  
  ),
```

File: keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+   * | Shift|   !  |   @  |   #  |   $  |   %  |  GUI |                    |  GUI |   _  |   +  |   <  |   >  |   ?  | Shift|
+   * |-------------+------+------+------+------+------+------+------+------+------+------+------+------+------+-------------|
+   * | Ctrl |  ESC |  ALt | EISU |||||||| kana | Raise| Space|||||||| Enter| Lower|Delete|||||||| Home |PageDn|PageUp|  End |
+   * ,----------------------------------------------------------------------------------------------------------------------.
+  */
+  [_RAISE] = LAYOUT(
+    KC_GRV , KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_UNDS,                       KC_PLUS, KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_PIPE, \
+    KC_LCTL, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_LCBR,                       KC_RCBR, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, KC_COLN, KC_DQT , \
+    KC_LSFT, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_LGUI,                       KC_RGUI, KC_UNDS, KC_PLUS, KC_LT,   KC_GT,   KC_QUES, KC_RSFT, \
+    KC_LCTL, KC_ESC,  KC_LALT, KC_LANG2,         KC_LANG1,RAISE,   KC_SPC,       KC_ENT, LOWER,   KC_DEL,           KC_HOME, KC_PGDN, KC_PGUP, KC_END   \
```
Comment (drashna) [inline keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c:87]:
```suggestion
    KC_GRV , KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_UNDS,                       KC_PLUS, KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_PIPE,
    KC_LCTL, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_LCBR,                       KC_RCBR, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, KC_COLN, KC_DQT ,
    KC_LSFT, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_LGUI,                       KC_RGUI, KC_UNDS, KC_PLUS, KC_LT,   KC_GT,   KC_QUES, KC_RSFT,
    KC_LCTL, KC_ESC,  KC_LALT, KC_LANG2,         KC_LANG1,RAISE,   KC_SPC,       KC_ENT, LOWER,   KC_DEL,           KC_HOME, KC_PGDN, KC_PGUP, KC_END  
```

File: keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+   * |-------------+------+------+------+------+------+------+------+------+------+------+------+------+------+-------------|
+   * |      |      |      |      ||||||||      |      |      ||||||||      |      |      ||||||||      |      |      |      |
+   * ,----------------------------------------------------------------------------------------------------------------------.
+   */
+  [_ADJUST] = LAYOUT(
+    _______, RESET  , RGB_TOG, RGB_MOD, RGB_HUD, RGB_HUI,_______,                       _______, RGB_SAD, RGB_SAI, RGB_VAD, RGB_VAI, _______, _______, \
+    _______, _______, BL_TOGG, BL_BRTG, BL_INC , BL_DEC ,_______,                       _______, _______, _______, _______, _______, _______, _______, \
+    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,    KC_F6 ,_______,                       _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+    _______, _______, _______, _______,          _______,_______,_______,       _______,_______, _______,          _______, _______, _______, _______  \
+  )
```
Comment (drashna) [inline keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c:106]:
```suggestion
    _______, RESET  , RGB_TOG, RGB_MOD, RGB_HUD, RGB_HUI,_______,                       _______, RGB_SAD, RGB_SAI, RGB_VAD, RGB_VAI, _______, _______,
    _______, _______, BL_TOGG, BL_BRTG, BL_INC , BL_DEC ,_______,                       _______, _______, _______, _______, _______, _______, _______,
    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,    KC_F6 ,_______,                       _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12, 
    _______, _______, _______, _______,          _______,_______,_______,       _______,_______, _______,          _______, _______, _______, _______ 
  )
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c (hunk 1 above)
Comment (ToyoshimaHidenori) [reply keyboards/ergodash/mini/keymaps/ToyoshimaHidenori/keymap.c]: Thank you for your code review!
I have fixed code according your suggestions in e5890bd.
Thank you.

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Review by drashna — APPROVED** [review]
Awesome, thanks!

Though, one last thing.  Could you make sure that the folder name is completely lower case?

**Comment by ToyoshimaHidenori** [comment]
> Awesome, thanks!
> 
> Though, one last thing. Could you make sure that the folder name is completely lower case?

I appreciate your review.

I changed the folder name from uppercase to lowercase. (adb450a)

**Comment by ToyoshimaHidenori** [comment]
I have removed the uppercase folder. (d285be7)

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**